pmd-core/src/main/java/net/sourceforge/pmd/util/IOUtil.java

```diff
@@ -169,4 +180,307 @@ public static void ensureClosed(List<? extends AutoCloseable> toClose,
         }
     }
 
+    public static void closeQuietly(Closeable closeable) {
```

Comment:
```suggestion

    // The following methods are taken from Apache Commons IO.
    // The dependency was removed from PMD 6 because it had a security issue,
    // and upgrading was not possible without upgrading to Java 8.
    // See https://github.com/pmd/pmd/pull/3968
    // TODO PMD 7: consider bringing back commons-io and cleaning this class up.

    public static void closeQuietly(Closeable closeable) {
```